code/datums/components/recharger.dm

```diff
@@ -0,0 +1,228 @@
+/datum/component/recharger
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	/// Ref to the inserted item, if any
+	var/obj/item/inserted_item
+	/// If we're currently recharging anything
+	var/recharging = FALSE
+	/// A typecache of what items can be inserted into the recharger
+	var/list/valid_items
+	/// A list of the default acceptable items to be inserted into the recharger, in case one isn't passed in during init
+	var/static/list/default_valid_items = list(
```

Comment:
is this neccessary? wouldn't cells be able to work on any /obj/item? hell with minor tweaks i'm sure it'd be applicable to synths

Reply:
It's not necessary, but I do think it's good to keep it this way for now. This allows for certain rechargers to only allow certain items, or for an item that can't be placed in a recharger

Reply:
indeed 👍